### PR TITLE
feat: Connect Create Token modal to network

### PR DESCRIPTION
**Pull request by stanleyjones**
Copy of https://github.com/liftedinit/gwen/pull/18 after the connection was lost and I had to delete my personal fork. Original description below.

---

## Description

See #17 for previous work. This PR _really_ starts from the second commit (https://github.com/liftedinit/gwen/pull/18/commits/73527ffc125331f73c873c812d83bb3805881987).

- [x] Submit form data to the network
- [x] Handle form submission errors
- [x] Redirect to Ledger Settings and re-fetch token list
- [ ] ~e2e test~ moved to future PR

## Related Issue

Closes liftedinit/roadmap#21

## Testing

Testing manually using the mock server. Will add e2e tests in the next PR, after the back-end is working so they pass.

## Breaking Changes

None

### CODE REVIEW

**Comment by stanleyjones** [comment]
This has now been tested using the mock server.

File: src/features/network/network-provider.tsx [hunk 1]
```diff
@@ -7,39 +7,42 @@ import {
...
-    cmdNetwork.apply([Ledger, IdStore, Account])
-    return [queryNetwork, cmdNetwork] as [Network, Network]
-  }, [activeNetwork, activeAccount])
+    const anonIdentity = new AnonymousIdentity();
+    const identity = activeAccount?.identity ?? anonIdentity;
+    const url = activeNetwork?.url || "";
+    const queryNetwork = new Network(url, anonIdentity);
+    queryNetwork.apply([Ledger, IdStore, Account, Events, Base, Tokens]);
+    const cmdNetwork = new Network(url, identity);
+    cmdNetwork.apply([Ledger, IdStore, Account, Tokens]);
```
Comment (stanleyjones) [inline src/features/network/network-provider.tsx:35]: Modules have to be explicitly bound _in the client app_ to be accessed. I don't love this pattern and will work to address it in ManyJs.

File: src/features/network/network-provider.tsx [hunk 2]
```diff
@@ -7,39 +7,42 @@ import {
...
+    return [queryNetwork, cmdNetwork] as [Network, Network];
```
Comment (stanleyjones) [inline src/features/network/network-provider.tsx:36]: This is existing code and also (IMHO) a little weird. We create two different network connections, an anonymous one for queries and an authenticated one for mutations. Are queries ever going to be permissioned?

File: src/views/layout.tsx [hunk 3]
```diff
@@ -2,12 +2,27 @@ import React from "react";
...
+  RiBankCardLine,
+} from "react-icons/ri";
+
+const navItems = [
+  { label: "Dashboard", path: "", icon: RiHome2Line },
+  { label: "Services", path: "settings", icon: RiServerLine },
+  { label: "Flows", path: "flows", icon: RiFlowChart },
+  { label: "Users", path: "users", icon: RiUserLine },
+  { label: "Billing", path: "account", icon: RiBankCardLine },
+];
```
Comment (stanleyjones) [inline src/views/layout.tsx:19]: These have to be explicitly added now. They were in the Nav component before, but that component is now in the shared library and this config doesn't belong there.

File: src/features/network/network-provider.tsx (hunk 2 above)
Comment (hansl) [reply src/features/network/network-provider.tsx:36]: They could be.

File: src/features/network/network-provider.tsx (hunk 1 above)
Comment (stanleyjones) [reply src/features/network/network-provider.tsx:35]: Hmm... let me think about this a little more then. My gut says, if we're trying to treeshake and not supply the whole library, we should consider something more like this:

```js
import Ledger from "@liftedinit/many/ledger";

const ledger = new Ledger(url, identity);
const info = await ledger.info();
```

File: src/features/network/network-provider.tsx (hunk 2 above)
Comment (stanleyjones) [reply src/features/network/network-provider.tsx:36]: I don't want to reinvent everything that Tommy did just to "make it my own" or anything, but I do think it needs a second look from the perspective of the app developer. The JS library should make app development really simple and straightforward and I'm not 100% sure it does in its current form.

File: src/features/network/network-provider.tsx (hunk 1 above)
Comment (hansl) [reply src/features/network/network-provider.tsx:35]: Star re-exports are properly tree shaken by webpack, but you're right.

**Review by mjsorribas — APPROVED** [review]

**Pull request merged**